LiteCore/Query/SQLiteN1QLFunctions.cc

```diff
@@ -1220,6 +1234,46 @@ namespace litecore {
         }
     }
 
+    // toarray(v) returns
+    //   MISSING              if v == MISSING,
+    //   NULL                 if v == NULL
+    //   v                    if v is an array
+    //   wrapp v in an array  otherwise
+    static void toarray(sqlite3_context* ctx, int argc, sqlite3_value **argv) noexcept {
+        try {
+            if (isMissing(argv[0]) || isNull(argv[0]) || isArray(ctx, argv[0])) {
+                sqlite3_result_value(ctx, argv[0]);
+            } else {
+                Encoder enc;
+                enc.beginArray();
+                writeSQLiteArg(ctx, argv[0], enc);
+                enc.endArray();
+                setResultBlobFromFleeceData( ctx,  enc.finish() );
+            }
+        } catch (const std::exception &) {
+            sqlite3_result_error(ctx, "toarray: exception!", -1);
+        }
+    }
+
+    // toobject(v) returns
+    //   MISSING              if v == MISSING,
+    //   NULL                 if v == NULL
+    //   v                    if v is an object
+    //   {}                   (the empty object), otherwise
+    static void toobject(sqlite3_context* ctx, int argc, sqlite3_value **argv) noexcept {
+        try {
+            if (isMissing(argv[0]) || isNull(argv[0]) || isObject(ctx, argv[0])) {
+                sqlite3_result_value(ctx, argv[0]);
+            } else {
+                Encoder enc;
+                enc.beginDictionary();
+                enc.endDictionary();
```

Comment:
I wonder if we could have a const value of a pre-encoded empty dictionary instead of encoding it each time?  Although that's probably such a small gain that it won't be worth it.

Reply:
Will do it.